iceoryx_utils/include/iceoryx_utils/internal/cxx/string.inl

```diff
@@ -343,6 +407,192 @@ inline bool operator!=(const char* const, const string<Capacity>&)
+}
+
+template <uint64_t Capacity>
+template <typename T>
+inline typename std::enable_if<internal::IsCharArray<T>::value || internal::IsCxxString<T>::value, bool>::type
+string<Capacity>::unsafe_append(const T& t) noexcept
+{
+    uint64_t tSize = internal::GetSize<T>::call(t);
+    if (Capacity < (m_rawstringSize + tSize))
+    {
+        std::cerr << "Appending failed because the sum of sizes exceeds this' capacity." << std::endl;
+        return false;
+    }
+    std::memcpy(m_rawstring + m_rawstringSize, internal::GetData<T>::call(t), tSize);
+    m_rawstring[m_rawstringSize + tSize] = '\0';
+    m_rawstringSize += tSize;
+    return true;
+}
+
+template <uint64_t Capacity>
+template <typename T>
+inline
+    typename std::enable_if<internal::IsCharArray<T>::value || internal::IsCxxString<T>::value, string<Capacity>&>::type
+    string<Capacity>::append(TruncateToCapacity_t, const T& t) noexcept
+{
+    uint64_t tSize = internal::GetSize<T>::call(t);
+    const char* tData = internal::GetData<T>::call(t);
+    if (Capacity < (m_rawstringSize + tSize))
+    {
+        std::cerr << "The last " << tSize - Capacity + m_rawstringSize << " characters of " << tData
+                  << " are truncated, because the length is larger than the capacity." << std::endl;
+        std::memcpy(m_rawstring + m_rawstringSize, tData, Capacity - m_rawstringSize);
+        m_rawstring[Capacity] = '\0';
+        m_rawstringSize = Capacity;
+    }
+    else
+    {
+        std::memcpy(m_rawstring + m_rawstringSize, tData, tSize);
+        m_rawstring[m_rawstringSize + tSize] = '\0';
+        m_rawstringSize += tSize;
+    }
+    return *this;
+}
+
+template <uint64_t Capacity>
+inline iox::cxx::optional<string<Capacity>> string<Capacity>::substr(uint64_t pos, uint64_t count) const noexcept
+{
+    if (pos > m_rawstringSize)
+    {
+        return iox::cxx::nullopt;
+    }
+    if (m_rawstringSize < (pos + count))
+    {
+        count = m_rawstringSize - pos;
+    }
+    string subString;
+    std::memcpy(subString.m_rawstring, &m_rawstring[pos], count);
+    subString.m_rawstring[count] = '\0';
+    subString.m_rawstringSize = count;
+    return subString;
+}
+
+template <uint64_t Capacity>
+inline iox::cxx::optional<string<Capacity>> string<Capacity>::substr(uint64_t pos) const noexcept
+{
+    return substr(pos, m_rawstringSize);
+}
+
+template <uint64_t Capacity>
+template <typename T>
+inline typename std::enable_if<std::is_same<T, std::string>::value || internal::IsCharArray<T>::value
+                                   || internal::IsCxxString<T>::value,
+                               iox::cxx::optional<uint64_t>>::type
+string<Capacity>::find(const T& t, uint64_t pos) const noexcept
+{
+    if (pos > m_rawstringSize)
+    {
+        return iox::cxx::nullopt;
+    }
+    const char* found = std::strstr(c_str() + pos, internal::GetData<T>::call(t));
+    if (found == nullptr)
+    {
+        return iox::cxx::nullopt;
+    }
+    return (found - c_str());
```

Comment:
shouldn't pointer arithmetic be done with uintptr_t?

Reply:
then we would lose the type information and we need the size of the type for the pointer arithmetic